packages/ui/components/createButton/CreateButton.tsx

```diff
@@ -0,0 +1,128 @@
+
+  const CreateDialog = props.createDialog ? props.createDialog() : null;
+  const DuplicateDialog = props.duplicateDialog ? props.duplicateDialog() : null;
+
+  const hasTeams = !!props.options.find((option) => option.teamId);
+
+  // inject selection data into url for correct router history
+  const openModal = (option: Parent) => {
+    const query = {
+      ...router.query,
+      dialog: "new",
+      eventPage: option.slug,
+      teamId: option.teamId,
+    };
+    if (!option.teamId) {
+      delete query.teamId;
+    }
+    router.push(
+      {
+        pathname: router.pathname,
+        query,
+      },
+      undefined,
+      { shallow: true }
+    );
+  };
+
+  return (
+    <>
+      {!hasTeams ? (
+        <Button
+          onClick={() =>
+            !!CreateDialog
+              ? openModal(props.options[0])
+              : props.createFunction
+              ? props.createFunction(props.options[0].teamId || undefined)
+              : null
+          }
+          data-testid="new-event-type"
+          StartIcon={FiPlus}
+          loading={props.isLoading}
+          variant={props.disableMobileButton ? "button" : "fab"}>
+          {props.buttonText ? props.buttonText : t("new")}
+        </Button>
+      ) : (
+        <Dropdown>
+          <DropdownMenuTrigger asChild>
+            <Button
+              variant={props.disableMobileButton ? "button" : "fab"}
+              StartIcon={FiPlus}
+              loading={props.isLoading}>
+              {props.buttonText ? props.buttonText : t("new")}
+            </Button>
+          </DropdownMenuTrigger>
+          <DropdownMenuContent sideOffset={14} align="end">
+            <DropdownMenuLabel>
+              <div className="w-48 text-left text-xs">{props.subtitle}</div>
+            </DropdownMenuLabel>
+            {props.options.map((option) => (
+              <DropdownMenuItem key={option.slug}>
+                <DropdownItem
+                  type="button"
+                  StartIcon={(props: any) => (
+                    <Avatar
+                      alt={option.name || ""}
+                      imageSrc={option.image || `${WEBAPP_URL}/${option.slug}/avatar.png`} // if no image, use default avatar
+                      size="sm"
+                      {...props}
+                    />
+                  )}
+                  onClick={() =>
+                    !!CreateDialog
+                      ? openModal(option)
+                      : props.createFunction
+                      ? props.createFunction(option.teamId || undefined)
+                      : null
+                  }>
+                  {" "}
+                  {/*improve this code */}
+                  <span>{option.name ? option.name : option.slug}</span>
+                </DropdownItem>
+              </DropdownMenuItem>
+            ))}
+          </DropdownMenuContent>
+        </Dropdown>
+      )}
+      {router.query.dialog === "duplicate" && DuplicateDialog}
```

Comment:
With the above suggestion, this should be moved to onSelect/onClick callback(`createFn` can also be removed) and we should let the component user decide if they wan to show a duplicateDialog or create Dialog.